src/factoid/util.rs

```diff
@@ -0,0 +1,83 @@
+use std::iter::Peekable;
+
+// e.g. awoo -> awooooo or meow -> meoooow
+pub fn is_extension(base: &String, candidate: &String) -> bool {
```

Comment:
oop forgot about the `&String` thing here too, which forces you into a bit of extra work at call sites here too

Reply:
if you want to punt on the `&String` arguments i'll tweak these and force you to review it, beware